c7n/resources/cw.py

```
@@ -114,6 +119,135 @@ def get_dimensions(self, resource):
+        load_resources(
+            [
+                "aws.sqs",
+                "aws.lambda",
+                "aws.ecs-cluster",
+                "aws.ecs-task",
+                "aws.kinesis",
+                "aws.sns",
+                "aws.ssm-parameter",
+                "aws.batch-compute",
+                "aws.codepipeline",
+            ]
+        )
+        arn_resolver = ArnResolver(self.manager)
+        arn_map = self._get_arns(resources)
+        results = []
+        for k, v in arn_map.items():
+            resolved = arn_resolver.resolve(v['Arns'])
+            if not all(resolved.values()):
+                for i, j in resolved.items():
+                    if not j:
+                        v['data'].setdefault('c7n:InvalidTargets', []).append(i)
+                results.append(v['data'])
+        return results
+
+
+@EventRule.action_registry.register('delete')
+class EventRuleDelete(BaseAction):
+    """
+    Delete an event rule, force target removal with the `force` option
+
+    :example:
+
+    .. code-block:: yaml
+
+        policies:
+            - name: force-delete-rules
+              resource: aws.event-rule
+              filters:
+                - Name: my-event-rule
+              actions:
+                - type: delete
+                  force: true
+    """
+
+    schema = type_schema('delete', force={'type': 'boolean'})
+    permissions = ('events:DeleteRule', 'events:RemoveTargets', 'events:ListTargetsByRule',)
+
+    def process(self, resources):
+        client = local_session(self.manager.session_factory).client('events')
+        children = {}
+        target_error_msg = "Rule can't be deleted since it has targets."
+        for r in resources:
+            try:
+                client.delete_rule(Name=r['Name'])
+            except botocore.exceptions.ClientError as e:
+                if self.data.get('force') and e.response['Error']['Message'] == target_error_msg:
```

use if if != and then raise, else this except blocks swallows exception, plus lets the rest of this code block become dedented